skfem/mesh/mesh.py

```diff
@@ -349,13 +349,8 @@ def __post_init__(self):
         if self.sort_t:
             self.t = np.sort(self.t, axis=0)
 
-        if not isinstance(self.doflocs, ndarray):
-            # for backwards compatibility: support standard lists
-            self.doflocs = np.array(self.doflocs, dtype=np.float64)
-
-        if not isinstance(self.t, ndarray):
-            # for backwards compatibility: support standard lists
-            self.t = np.array(self.t, dtype=np.int64)
+        self.doflocs = np.asarray(self.doflocs, dtype=np.float64)
+        self.t = np.asarray(self.t, dtype=np.int64)
```

Comment:
Will this always convert the memory layout to `C_CONTIGUOUS`? Thus, making the checks below redundant?

Reply:
I think there is a keyword argument `order='K'` to keep the memory layout of the original array if this was not intended.

Reply:
Hmm, sorry, I have no idea. I've many times wondered what the two warnings I often get about C-contiguity are but not yet gotten around to looking into what they mean or how to avoid them. So, no, I didn't intend to change any behaviour here. Sorry. What's the right thing to do here?

Reply:
In scikit-fem we do lots of slices like `mesh.p[0, :]` which is more cache-efficient if those numbers are next to each other in memory. So the conversion basically reorders those numbers so that `mesh.p[0, 0]` is just before `mesh.p[0, 1]` which is just before `mesh.p[0, 2]` and so on. However, because this has nonzero computational cost, I wanted to warn the user: sometimes this can be fixed outside scikit-fem.

Reply:
Now this PR does the conversion silently, without the warning. I'd like to preserve the warning so if you add `order='K'` no reordering will be done by `np.asarray`.

Reply:
Btw, I'm not sure if cache-efficiency is the correct term, but what I'm trying to say is that these CPU's will assume a specific access order and preload certain numbers to L1 cache.

Reply:
Right. That makes a lot of sense and sounds important.  I'll make the suggested change when I get to a terminal.

Could you elaborate on

> sometimes this can be fixed outside scikit-fem.

?

I have seen and ignored a lot of these warnings; have I been passing low-hanging fruit by?